File: modules/crypto-core/src/main/java/com/shterneregen/securelan/crypto/service/FileCryptoService.java
package com.shterneregen.securelan.crypto.service;

import com.shterneregen.securelan.crypto.model.HybridEncryptedData;
import com.shterneregen.securelan.crypto.model.PasswordEncryptedData;

import java.nio.file.Path;
import java.security.PrivateKey;
import java.security.PublicKey;

public interface FileCryptoService {
    HybridEncryptedData encryptFileWithPublicKey(Path path, PublicKey publicKey);

    byte[] decryptFileWithPrivateKey(HybridEncryptedData encryptedData, PrivateKey privateKey);

    PasswordEncryptedData encryptFileWithPassword(Path path, char[] password);

    byte[] decryptFileWithPassword(PasswordEncryptedData encryptedData, char[] password);
}

File: modules/crypto-core/src/main/kotlin/com/shterneregen/securelan/crypto/service/FileCryptoService.kt
package com.shterneregen.securelan.crypto.service

import com.shterneregen.securelan.crypto.model.HybridEncryptedData
import com.shterneregen.securelan.crypto.model.PasswordEncryptedData
import java.nio.file.Path
import java.security.PrivateKey
import java.security.PublicKey

interface FileCryptoService {
    fun encryptFileWithPublicKey(path: Path, publicKey: PublicKey): HybridEncryptedData

    fun decryptFileWithPrivateKey(encryptedData: HybridEncryptedData, privateKey: PrivateKey): ByteArray

    fun encryptFileWithPassword(path: Path, password: CharArray): PasswordEncryptedData

    fun decryptFileWithPassword(encryptedData: PasswordEncryptedData, password: CharArray): ByteArray
}
